main.go

```diff
@@ -421,6 +432,61 @@ func (h *Handler) handle(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *Handler) handle(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Request from %s at \"%s\"", r.RemoteAddr, r.Header.Get("Origin"))
+
+	// Set the CORS headers
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Methods", "POST")
+	w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token")
+
+	// Handle preflight request (CORS)
+	switch r.Method {
+	case "OPTIONS":
+		w.WriteHeader(http.StatusOK)
+		return
+	case "POST":
+		var sfuAccessRequest SFURequest
+
+		decoder := json.NewDecoder(r.Body)
+		decoder.DisallowUnknownFields()
+		if err := decoder.Decode(&sfuAccessRequest); err == nil {
+			if err := sfuAccessRequest.Validate(); err != nil {
+				matrixErr := &MatrixErrorResponse{}
+				if errors.As(err, &matrixErr) {
+					log.Printf("Error processing request: %v", matrixErr.Err)
+					writeMatrixError(w, matrixErr.Status, matrixErr.ErrCode, matrixErr.Err)
+					return
+				}
+			}
+		} else {
+            log.Printf("Error reading request body: %v", err)
+            writeMatrixError(w, http.StatusBadRequest, "M_NOT_JSON", "Error reading request")
+            return
+        }			
+
+		log.Printf("Processing SFU request")
+		sfuAccessResponse, err := h.processSFURequest(r, &sfuAccessRequest)
+
+		if err != nil {
+			matrixErr := &MatrixErrorResponse{}
+			if errors.As(err, &matrixErr) {
+				log.Printf("Error processing request: %v", matrixErr.Err)
+				writeMatrixError(w, matrixErr.Status, matrixErr.ErrCode, matrixErr.Err)
+				return
+			}
+		}
+
+        w.Header().Set("Content-Type", "application/json")
```

Comment:
Is this in the right place, presumably the matrix errors are also JSON. Do we ever respond with anything not-JSON?